react/src/tests/opReturn.test.ts

```diff
@@ -85,94 +125,102 @@ describe('getDataPushdata', () => {
-  it('Empty data no paymentId', () => {
-    const fullResult = parseOpReturnProps('', true)
-    expect(fullResult).toBe(allResultsPrefix + '00') // 00 is pushdata for no data
-  })
-  it('Simple parse OpReturn 1', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData') // 16 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
-  it('Simple parse OpReturn 1 no paymentId', () => {
-    const fullResult = parseOpReturnProps('myCustomUserData', true) // 16 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '10' + // 16 in hex
-      '6d79437573746f6d5573657244617461')
-  })
-  it('Simple parse OpReturn 2', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data') // 33 bytes
-    const resultPaymentId = fullResult?.slice(-paymentIdDigits)
-    const resultNoPaymentId = fullResult?.slice(0,-paymentIdDigits)
-    expect(resultNoPaymentId).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-    expect(resultPaymentId?.length).toBe(paymentIdDigits)
-    expect(resultPaymentId).toMatch(paymentIdRegex)
-  })
-  it('Simple parse OpReturn 2 no paymentId', () => {
-    const fullResult = parseOpReturnProps('my=Longer more=sensible|user|data', true) // 33 bytes
-    expect(fullResult).toBe(allResultsPrefix +
-      '21' + // 33 in hex
-      '6d793d4c6f6e676572206d6f72653d73656e7369626c657c757365727c64617461')
-  })
-  it('Throws if too long', () => {
-    const data208bytes = '😂'.repeat(52) // 52 * 4 = 208 bytes
-    expect(() =>
-      parseOpReturnProps(data208bytes)
-    ).toThrow('Maximum 205 byte size exceeded for user data: 208')
-  })
-  it('Is not too long if no paymentId', () => {
-    const data208bytes = '😂'.repeat(52) // 52 * 4 = 208 bytes
-    const fullResult = parseOpReturnProps(data208bytes, true)
-    expect(fullResult).toBe(allResultsPrefix +
-      '4cd0' + // 4c because pushData > 75; d0 is 208 in hex
-      'f09f9882'.repeat(52)
-    )
-   })
-  it('Throws if too long no paymentId', () => {
-    const data216bytes = '😂'.repeat(54) // 54 * 4 = 216 bytes
-    expect(() =>
-      parseOpReturnProps(data216bytes, true)
-    ).toThrow('Maximum 214 byte size exceeded for user data: 216')
-  })
-})
+  const allResultsPrefix = '04' + '50415900' + '00';
+  const paymentIdDigits = 18;
+  const paymentIdRegex = /^08[0-9a-fA-F]{16}$/;
+  it('Encodes undefined data with paymentId', () => {
+    const fullResult = encodeOpReturnProps(undefined);
+    const resultData = fullResult?.slice(0, -paymentIdDigits);
+    expect(resultData).toBe(allResultsPrefix + '00'); // 00 is pushdata for no data
```

Comment:
Would help to have a comment above this line, something like

// Test the string through the data push, as paymentId is expected to be random every time